code/game/machinery/exonet_node.dm

```diff
@@ -0,0 +1,130 @@
+/obj/machinery/exonet_node
+	name = "exonet node"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "exonet_node"
+	idle_power_usage = 25
+	var/on = TRUE
+	var/toggle = 1
+	density = TRUE
+	anchored = TRUE
+	circuit = /obj/item/circuitboard/machine/exonet_node
+	max_integrity = 300
+	integrity_failure = 100
+	armor = list("melee" = 20, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 50, "acid" = 70)
+	desc = "This machine is exonet node."
+	var/list/logs = list() // Gets written to by exonet's send_message() function.
+	var/opened = FALSE
+
+// Proc: New()
+// Parameters: None
+// Description: Adds components to the machine for deconstruction.
+/*
+/obj/machinery/exonet_node/map/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/machine/exonet_node(src)
+	component_parts += new /obj/item/stock_parts/subspace/ansible(src)
+	component_parts += new /obj/item/stock_parts/subspace/filter(src)
+	component_parts += new /obj/item/stock_parts/manipulator(src)
+	component_parts += new /obj/item/stock_parts/manipulator(src)
+	component_parts += new /obj/item/stock_parts/micro_laser(src)
+	component_parts += new /obj/item/stock_parts/subspace/crystal(src)
+	component_parts += new /obj/item/stock_parts/subspace/treatment(src)
+	component_parts += new /obj/item/stock_parts/subspace/treatment(src)
+	component_parts += new /obj/item/stack/cable_coil(src, 2)
+	RefreshParts()
+*/
+
+
+// Proc: update_icon()
+// Parameters: None
+// Description: Self explanatory.
+/obj/machinery/exonet_node/update_icon()
+	if(on)
+		icon_state = initial(icon_state)
+	else
+		icon_state = "[initial(icon_state)]_off"
+
+// Proc: update_power()
+// Parameters: None
+// Description: Sets the device on/off and adjusts power draw based on stat and toggle variables.
+/obj/machinery/exonet_node/proc/update_power()
+	if(toggle)
+		if(!is_operational())
+			on = FALSE
+		else
+			on = TRUE
+	else
+		on = FALSE
+	use_power = on
+	update_icon()
+
+// Proc: emp_act()
+// Parameters: 1 (severity - how strong the EMP is, with lower numbers being stronger)
+// Description: Shuts off the machine for awhile if an EMP hits it.  Ion anomalies also call this to turn it off.
+/obj/machinery/exonet_node/emp_act(severity)
+	if(!(stat & EMPED))
+		stat |= EMPED
+		var/duration = (300 * 10)/severity
```

Comment:
a severity 1 emp will shut it off for 6 minutes are you sure this is the same as rest of telecoms..?

Reply:
Severity 1 is strong, 2 is weak. Strong EMP duration is 5 minutes, weak is 2 1/2 minutes. Tcomms has a +- 2 second variance though. It is otherwise the same as other Tcomms equipment.

Reply:
alright.